crates/lunatic-networking-api/src/tls_tcp.rs

```diff
@@ -0,0 +1,878 @@
+use std::convert::TryInto;
+use std::future::Future;
+use std::io::{self, IoSlice};
+use std::sync::Arc;
+use std::time::Duration;
+
+use anyhow::Result;
+use tokio::time::timeout;
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::{TcpListener, TcpStream},
+};
+use wasmtime::Trap;
+use wasmtime::{Caller, Linker};
+
+use lunatic_common_api::{get_memory, IntoTrap};
+use lunatic_error_api::ErrorCtx;
+
+use crate::dns::DnsIterator;
+use crate::{socket_address, NetworkingCtx, TlsConnection, TlsListener};
+use tokio_rustls::rustls::{self, OwnedTrustAnchor};
+use tokio_rustls::{webpki, TlsAcceptor, TlsConnector, TlsStream};
+
+// Register TCP networking APIs to the linker
+pub fn register<T: NetworkingCtx + ErrorCtx + Send + 'static>(
+    linker: &mut Linker<T>,
+) -> Result<()> {
+    linker.func_wrap10_async("lunatic::networking", "tls_bind", tls_bind)?;
+    linker.func_wrap(
+        "lunatic::networking",
+        "drop_tls_listener",
+        drop_tls_listener,
+    )?;
+    linker.func_wrap("lunatic::networking", "tls_local_addr", tls_local_addr)?;
+    linker.func_wrap3_async("lunatic::networking", "tls_accept", tls_accept)?;
+    linker.func_wrap7_async("lunatic::networking", "tls_connect", tls_connect)?;
+    linker.func_wrap("lunatic::networking", "drop_tls_stream", drop_tls_stream)?;
+    linker.func_wrap("lunatic::networking", "clone_tls_stream", clone_tls_stream)?;
+    linker.func_wrap4_async(
+        "lunatic::networking",
+        "tls_write_vectored",
+        tls_write_vectored,
+    )?;
+    // linker.func_wrap4_async("lunatic::networking", "tls_peek", tls_peek)?;
```

Comment:
Do we need peek timeouts if peek doesn't work for now? I would also remove this comment.

Reply:
removed all the peek related functionality